karapace/protobuf/schema.py

```diff
@@ -1,14 +1,188 @@
 # Ported from square/wire:
 # wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Schema.kt
 # Ported partially for required functionality.
+from karapace.protobuf.enum_element import EnumElement
+from karapace.protobuf.location import Location
+from karapace.protobuf.message_element import MessageElement
+from karapace.protobuf.option_element import OptionElement
+from karapace.protobuf.proto_file_element import ProtoFileElement
+from karapace.protobuf.proto_parser import ProtoParser
+from karapace.protobuf.utils import append_documentation, append_indented
 
+import logging
 
+log = logging.getLogger(__name__)
+
+
+def static_init(cls):
+    if getattr(cls, "static_init", None):
+        cls.static_init()
+    return cls
+
+
+def add_slashes(text: str):
```

Comment:
```suggestion
def add_slashes(text: str) -> str:
```

---

`r'\a'` is short-hand for `'\\a'`

> Both string and bytes literals may optionally be prefixed with a letter 'r' or 'R'; such strings are called raw strings and treat backslashes as literal characters

https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals

So the keys from `r'\a'` to `r'\v'` in `escape_dict` are actually two code-point long Unicode strings:

```python
>>> len(r'\a')
2
```

Because the loop `for char in text:` goes over a single code-point, i.e. its length is 1, those keys are never used. Either those can be removed or the code is not working as intended.

---

The `str` type in python has a built-in method to perform translations:

```python
>>> escape_quotes_and_backslash = str.maketrans({"'": "\\'", '"': '\\"', '\\': '\\\\'})
>>> "'".translate(escape_quotes_and_backslash)
"\\'"
```

Reply:
ok